File: frontend/src/types/transaction.ts
import type { Subscription } from './subscription';
import type { User } from './user';
import type { Plan } from './plan'; // Возможно, понадобится

export interface Transaction {
    id: number;
    user_id: number;
    amount: number;
    currency: string;
    status: 'completed' | 'pending' | 'failed';
    subscription_id: number | null;
    created_at: string;
    user: User;
    // Наша "жадная" загрузка в subscription включает и plan
    subscription: (Subscription & { plan: Plan }) | null;
    plan_id: number | null; // Мы добавили это поле
}

File: backend/app/models/transaction.py
from sqlalchemy import Column, Integer, ForeignKey, Float, DateTime, String
from sqlalchemy.orm import relationship
from backend.app.db.base_class import Base
from datetime import datetime, timezone


class Transaction(Base):
    __tablename__ = "transactions"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    subscription_id = Column(Integer, ForeignKey("subscriptions.id"), nullable=True)
    plan_id = Column(Integer, ForeignKey("plans.id"), nullable=True)
    amount = Column(Float, nullable=False)
    currency = Column(String, default="USD")
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    status = Column(String, default="completed")  # pending, completed, failed

    user = relationship("User")
    subscription = relationship("Subscription")
